File: python/bee1040.py
n1, n2, n3, n4 = [float(x) for x in input().split(' ')]

media = ((n1*2) + (n2*3) + (n3*4) + (n4*1)) / 10
print(f'Media: {media:.1f}')

if media >= 7:
    print('Aluno aprovado.')
elif 5 <= media < 7:
    print('Aluno em exame.')
    n5 = float(input())
    print(f'Nota do exame: {n5:.1f}')
    nova_media = (media + n5) / 2
    if nova_media >= 5.0:
        print('Aluno aprovado.')
    else:
        print('Aluno reprovado.')
    print(f'Media final: {nova_media:.1f}')
elif media < 5:
    print('Aluno reprovado.')

File: javascript/bee1040.js
var input = require("fs").readFileSync("stdin", "utf8");
var lines = input.split("\n");

const pesos = [2, 3, 4, 1];
const notas = lines[0].split(" ").map(Number);

let avg = notas.reduce((sum, nota, index) => sum + nota * pesos[index], 0) / pesos.reduce((total, peso) => total + peso, 0);
console.log("Media: " + avg.toFixed(1));

let sit = avg < 5.0 ? "Aluno reprovado." : avg <= 6.9 ? "Aluno em exame." : "Aluno aprovado.";
console.log(sit);

if(lines[1] !== undefined && lines[1] === "") {
    const notaFinal = Number(lines[1]);
	console.log("Nota do exame: " + notaFinal.toFixed(1));

	avg = (avg + notaFinal) / 2;
	sit = avg < 5.0 ? "Aluno reprovado." : "Aluno aprovado.";
	console.log(sit);
	console.log("Media final:", avg.toFixed(1));
}